frontend/views/containers/GroupPledgesGraph.vue

```diff
@@ -1,218 +1,204 @@
 
-      if (neededPledges) {
+      // Note: surplus is added on the innerSlices, so we need to substract its part from the pledges.
+      // To be fair, we remove the equivalent percentage of each pledge part (others and ours)
+
+      const slices = []
+
+      if (othersPledgesAmount > 0) {
+        const pledgePerc = (othersPledgesAmount / pledgeTotal).toFixed(2)
+        const surplusToBeRemoved = surplus * pledgePerc
+
         slices.push({
-          id: 'neededPledges',
-          percent: this.decimalSlice(neededPledges),
-          color: 'light', // TODO later - review this color, too light
-          label: this.L('{amount} needed pledge', { amount: currency + neededPledges })
+          id: 'othersPledgesAmount',
+          percent: this.decimalSlice(othersPledgesAmount - surplusToBeRemoved),
+          color: 'primary'
         })
-      } else if (surplus) {
+      }
+
+      if (ourPledgeAmount > 0) {
+        const pledgePerc = (ourPledgeAmount / pledgeTotal).toFixed(2)
+        const surplusToBeRemoved = surplus * pledgePerc
+
         slices.push({
-          id: 'surplus',
-          percent: this.decimalSlice(surplus),
-          color: 'secondary',
-          label: this.L('{amount} extra pledge', { amount: currency + surplus })
+          id: 'ourPledgeAmount',
+          percent: this.decimalSlice(ourPledgeAmount - surplusToBeRemoved),
+          color: 'primary'
         })
       }
 
       return slices
     },
-    groupPledgingInnerSlices () {
-      return [
-        {
-          id: 'userIncomeToReceive',
-          percent: this.decimalSlice(this.graphData.userIncomeToReceive),
-          color: 'tertiary'
-        }
-      ]
+    innerSlices () {
+      const { ourIncomeToReceive, surplus } = this.graphData
+
+      if (ourIncomeToReceive > 0) {
+        return [{
+          id: 'ourIncomeToReceive',
+          percent: this.decimalSlice(ourIncomeToReceive),
+          color: 'warning'
+        }]
+      }
+
+      if (surplus > 0) {
+        return [{
+          id: 'surplus',
+          percent: this.decimalSlice(surplus),
+          color: 'success'
+        }]
+      }
+
+      return []
     }
   },
   methods: {
-    decimalSlice (amount) {
-      // NOTE: When bigger than (group) goal, take in account the surplus slice
-      const { goal, totalAmount, surplus } = this.graphData
-      const circleCompleteAmount = Math.max(totalAmount + surplus, goal)
-      return amount / circleCompleteAmount
+    withCurrency (amount) {
+      return currencies[this.groupSettings.mincomeCurrency].displayWithCurrency(amount)
     },
-    handleResize: debounce(function () {
-      this.isMobile = this.verifyIsMobile()
-    }, 100),
-    verifyIsMobile () {
-      return window.innerWidth < TABLET
+    decimalSlice (amount) {
+      const perc = amount / this.graphData.groupGoal
+      // avoid breaking the graph when perc is bigger than 1,
+      if (perc > 1) return 1
+      if (perc < 0) return 0
+      return perc
```

Comment:
can save some lines:

```
return Math.max(1, Math.min(0, perc))
```

Reply:
This is wrong @taoeffect , that's what causing the bug you are facing (missing the white slice)

Reply:
Oops, it should have been:

```js
return Math.min(1, Math.max(0, perc))
```

Reply:
It makes sense now (and it works!) changed